src/lib/components/Form/PublishPanel.svelte

```diff
@@ -0,0 +1,206 @@
+<script lang="ts">
+  import type { MetadataCollection } from '$lib/models/metadata';
+  import { env } from '$env/dynamic/public';
+  import Button, { Icon, Label } from '@smui/button';
+  import Paper from '@smui/paper';
+  import Spinner from '../Spinner.svelte';
+
+  type MetadataPublishResponse = {
+    publishedCatalogRecords: string[];
+  };
+
+  type PublishPanelProps = {
+    metadata?: MetadataCollection;
+  };
+
+  let { metadata }: PublishPanelProps = $props();
+
+  let isLoading = $state(false);
+  let responseStatus = $state<number | null>(null);
+  let responseUuids = $state<MetadataPublishResponse | null>(null);
+
+  const isAllowedToPublish = $derived(() => {
+    return metadata?.approved === true && metadata?.responsibleRole === 'MdeEditor';
+  });
+
+  const onPublishClick = async () => {
+    if (!metadata?.metadataId) {
+      console.error('No metadata ID available to publish');
+      return;
+    }
+
+    try {
+      isLoading = true;
+      responseStatus = null;
+
+      const response = await fetch(`/metadata/${metadata.metadataId}/publish`, {
+        method: 'POST'
+      });
+
+      responseStatus = response.status;
+
+      if (!response.ok) {
+        throw new Error(`Failed to publish the metadata: ${response.statusText}`);
+      }
+
+      const uuids = (await response.json()) as MetadataPublishResponse;
+
+      responseUuids = uuids;
+    } catch (error) {
+      console.error('Error during publishing: ', error);
+    } finally {
+      isLoading = false;
+    }
+  };
+</script>
+
+<div class="publish-panel-container">
+  <Paper elevation={6} class="publish-panel">
```

Comment:
I'd prefer to have this as a `Dialog`. Compare `AssignmentDialog.svelte`.

Reply:
Agreed!

Will be fixed application-wide in a follow up.